test/extensions/filters/listener/proxy_protocol/proxy_proto_integration_test.cc

```diff
@@ -338,6 +338,12 @@ TEST_P(ProxyProtoFilterChainMatchIntegrationTest, MatchDirectSourceAndSource) {
               testing::HasSubstr("directsource_localhost_and_source_1.2.3.0/24 -"));
 }
 
+TEST_P(ProxyProtoFilterChainMatchIntegrationTest, CloseAfterProxyProtocol) {
+  send("PROXY TCP4 1.2.3.4 254.254.254.254 12345 1234", true);
```

Comment:
Using the `end_stream` flag to end the connection before sending the full proxy header. @ggreenway